spec/reek/smells/unused_private_method_spec.rb

```diff
@@ -0,0 +1,140 @@
+require_relative '../../spec_helper'
+require_lib 'reek/smells/unused_private_method'
+require_lib 'reek/examiner'
+require_relative 'smell_detector_shared'
+
+RSpec.describe Reek::Smells::UnusedPrivateMethod do
+  let(:detector) { build(:smell_detector, smell_type: :UnusedPrivateMethod) }
+
+  it_should_behave_like 'SmellDetector'
+
+  context 'unused private methods' do
+    it 'reports instance methods' do
+      source = <<-EOF
+        class Car
+          private
+          def start_the_engine; end
+          def drive; end
+        end
+      EOF
+
+      expect(source).to reek_of(:UnusedPrivateMethod, name: :start_the_engine)
+      expect(source).to reek_of(:UnusedPrivateMethod, name: :drive)
+    end
+
+    it 'reports class methods' do
+      source = <<-EOF
+        class Car
+          private
```

Comment:
This should be `private_class_method`, otherwise `Car.start_the_engine` is actually public.